### PR TITLE
updated SecurityConfig.java to include making device_manager page res…

### PR DESCRIPTION
**Pull request by Bsbuckne**
…tricted without logging in

### CODE REVIEW

**Comment by Bsbuckne** [comment]
This needs to be merged before I can test it on master, it works fine on the device manager branch. Check it out and if it's cool merge it and I'll QA it tomorrow

**Comment by cit280superepicness** [comment]
Conflicts with master resolved.

File: src/main/resources/templates/index.html [hunk 1]
```diff
@@ -0,0 +1,52 @@
+<!DOCTYPE html>
+<html xmlns:th="http://www.thymeleaf.org" xmlns:sec="http://www.thymeleaf.org/thymeleaf-extras-springsecurity4">
+<head>
+    <!-- Theme Made By www.w3schools.com - No Copyright -->
```
Comment (jrmullen) [inline src/main/resources/templates/index.html:4]: remove comment

File: src/main/resources/templates/index.html [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+          href="http://cdn.jsdelivr.net/webjars/bootstrap/3.3.1/css/bootstrap.css" rel="stylesheet" media="screen"/>
+
+</head>
+<body>
+
+<!-- Navbar -->
+<div th:include="fragments/header :: indexHeader"></div>
+
+
+<!-- First Container -->
```
Comment (jrmullen) [inline src/main/resources/templates/index.html:17]: remove comment

File: src/main/resources/templates/index.html [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+<div class="container-fluid bg-1 text-center">
+    <img src="http://i67.tinypic.com/2d0uw8.jpg" class="img-responsive" style="display:inline" alt="camera" width="450" height="450" />
+</div>
+
+<!-- Second Container (Grid) -->
```
Comment (jrmullen) [inline src/main/resources/templates/index.html:22]: remove comment

File: src/main/java/com/watchdog/config/SecurityConfig.java [hunk 4]
```diff
@@ -52,8 +53,8 @@ public void configureGlobal(AuthenticationManagerBuilder auth) throws Exception
                 .jdbcAuthentication().dataSource(dataSource)
                 .passwordEncoder(passwordEncoder())
                 .usersByUsernameQuery("SELECT USER_EMAIL, USER_PASSWORD, enabled FROM user WHERE USER_EMAIL = ?")
-                .authoritiesByUsernameQuery("select user.USER_EMAIL, permissions.ROLE from user, permissions "
-                                                            + "where user.USER_EMAIL = ?");
+                .authoritiesByUsernameQuery("select user.USER_EMAIL, permissions.ROLE from user, permissions " +
+                        "where permissions.PERMISS_ID = user.PERMISS_ID and user.USER_EMAIL = ?");
```
Comment (jrmullen) [inline src/main/java/com/watchdog/config/SecurityConfig.java:57]: New query has been tested, correct?

**Comment by cit280superepicness** [comment]
Yes, the new query has been tested. I am not able to get to my computer to
remove the comments till much later today so if you could remove the
comments yourself or just go ahead and merge so that Briana can watch test
it today like she mentioned y, that would be great.

**Comment by cit280superepicness** [comment]
I will look into it. I also found a workaround for
sec:authorize="hasRole('ROLE_ADMIN')" not working. I am going to see if I
can get it working on the device_manager branch. I will resolve the merge
conflicts as well.

I'll let you know if I am successful/when I am done.

**Comment by cit280superepicness** [comment]
The header is now only showing content I should for users with role_admin
and role_user.

**Comment by Bsbuckne** [comment]
I already fixed the device manager issue with the roles, I just haven't pushed it yet

**Comment by cit280superepicness** [comment]
Briana, a bunch of fixes related to roles were already pushed to the main
branch this morning. what specific issue with roles are you referring to?

**Comment by Bsbuckne** [comment]
I was still able to see the link to and get to the device_manager when logged in as a user. I fixed it to where that wasn't the case anymore

**Comment by cit280superepicness** [comment]
I fixed this and tested it on my system before and then pushed it to the
device manager branch. Unless someone saved over what I pushed before the
device manager was merged with master and that is the issue, did you try
doing "git pull origin master" or "git pull origin device_manager" to get
the most recent changes? Yo should be able to pull all the fixes.

Also, did you import the most recent version of the db file I uploaded
today? I repopulated the users table a few days ago to  have 1/2 the users
be admins and be normal users. Then today I made another change to the
script, I made it by default so that when you register as a  new user in
Watchdog, the user is a regular user and not an admin. If you are having
issues with a newly registered us er being able to access the
device_manager, not having the most recent script is why.

**Comment by cit280superepicness** [comment]
Hey Briana,

So I just looked at the device manager branch and everything is fine. Sorry
for the confusion on my part. I also see now you were right about that line
to restrict access based on user role for device_manager missing in the
SecurityConfig file.Thanks for noticing that and adding it!

Renee

**Comment by Bsbuckne** [comment]
I made my fixes to the device manager after we presented today and no I haven't imported your most recent script. I haven't touched our project since we left class earlier. I'll do another pull on Thursday. I'll be working on my other two class projects tonight and tomorrow

**Pull request merged**